### PR TITLE
add esp32 build document and ci  Refs #5536

### PR DESCRIPTION
**Pull request by luxincn**
add esp32 build doc and ci Refs #5536

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=5567) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=5567) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/ncnn?pullRequest=5567) it.</sub>

File: .github/workflows/esp32.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    branches: [master]
+    paths:
+      - '.github/workflows/esp32.yml'
+      - 'CMakeLists.txt'
+      - 'cmake/**'
+      - 'src/*'
+      - 'src/layer/*'
+      - 'src/layer/x86/**'
+      - 'src/layer/vulkan/**'
+      - 'tests/**'
```
Comment (nihui) [inline .github/workflows/esp32.yml:13]: I don't think esp32 has any vulkan support atm
and esp32 is not x86, and there is not any tests built

delete these lines

File: .github/workflows/esp32.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
```
Comment (nihui) [inline .github/workflows/esp32.yml:24]: same as above

File: .github/workflows/esp32.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+          path: esp-idf-install
+          key: esp-idf-${{ runner.os }}-install
+
+      - name: Clone and Install ESP-IDF
+        if: steps.Cache-ESP-IDF.outputs.cache-hit != 'true'
+        run: |
+          git clone https://github.com/espressif/esp-idf
+          cd esp-idf
+          git submodule update --init --recursive
+          ./install.sh
```
Comment (nihui) [inline .github/workflows/esp32.yml:66]: why do we cache the idf for dev branch ?
use action/checkout for this job, with idf release version but not the dev branch

File: docs/how-to-build/how-to-build.md [hunk 4]
```diff
@@ -866,3 +866,50 @@ ${HM_SDK}/native/build-tools/cmake/bin/cmake -DOHOS_STL=c++_static -DOHOS_ARCH=a
...
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}" CACHE STRING "c flags")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}" CACHE STRING "c++ flags")
+
+option(NCNN_BUILD_BENCHMARK "" OFF)
+```
```
Comment (nihui) [inline docs/how-to-build/how-to-build.md:902]: we have in-house toolchain file now, this section is not needed, right ?

Comment (luxincn) [reply docs/how-to-build/how-to-build.md]: Right, we can directly use the provided in the project

**Review by nihui — APPROVED** [review]

**Comment by nihui** [comment]
Thanks for your contribution !

**Pull request merged**